bofire/data_models/strategies/predictives/botorch.py

```diff
@@ -96,11 +96,14 @@ def update_surrogate_specs_for_domain(cls, values):
                         )
         return values
 
-    @model_validator(mode="after")
+    @model_validator(mode="before")
     @classmethod
     def validate_outlier_detection_specs_for_domain(cls, values):
         """Ensures that a outlier_detection model is specified for each output feature"""
-        if values["outlier_detection_specs"] is not None:
+        if (
+            "outlier_detection_specs" in values
+            and values["outlier_detection_specs"] is not None
```

Comment:
@jduerholt, here again.

Reply:
I only can say that it has worked :D

Reply:
My question was if the change is correct or if it changes what happens :D